src/main/java/com/ss/training/lms/controller/AdminGenreController.java

```diff
@@ -0,0 +1,67 @@
+package com.ss.training.lms.controller;
+
+import java.sql.SQLException;
+
+import com.ss.training.lms.entity.Genre;
+import com.ss.training.lms.service.admin.AdminGenreService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+public class AdminGenreController {
+	
+	@Autowired
+	AdminGenreService genreService;
+	
+	@RequestMapping(path="/lms/admin/genre/{genreId}", 
+					produces = {
+						MediaType.APPLICATION_XML_VALUE,
+						MediaType.APPLICATION_JSON_VALUE
+					})
+	public ResponseEntity<Genre>getGenreById(@PathVariable int genreId) throws SQLException {
+		Genre genre= genreService.readAGenre(genreId);
+		return new ResponseEntity<Genre>(genre, HttpStatus.CREATED);
```

Comment:
The resonse code should not be "created" here because this is a get method right.

Reply:
good catch!